sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1084,6 +1084,22 @@ object functions {
   def log(columnName: String): Column = log(Column(columnName))
 
   /**
+   * Returns the first argument-base logarithm of the second argument.
+   *
+   * @group math_funcs
+   * @since 1.4.0
+   */
+  def log(base: Double, a: Column): Column = Logarithm(lit(base).expr, a.expr)
+
+  /**
+   * Returns the first argument-base logarithm of the second argument.
+   *
+   * @group math_funcs
+   * @since 1.4.0
+   */
+  def log(base: Double, a: String): Column = log(base, Column(a))
+
```

Comment:
It is suggested by @rxin. I think it is reasonable because it is hard to have a use case to returns the logarithm of one column with another column as base. Usually you want to compute the logarithm values for a column with the same base.